src/config.ts

```diff
@@ -451,19 +459,30 @@ function loadPlugins(
   // if no mounted directories, mount root
   if (!Object.keys(mountedDirs).length) mountedDirs['.'] = '/';
 
-  const needsDefaultPlugin = new Set(['.js', '.mjs', '.jsx', '.ts', '.tsx']);
+  const needsDefaultPlugin = new Set(['.mjs', '.jsx', '.ts', '.tsx']);
   plugins
-    .reduce((arr, a) => arr.concat(a.input), [] as string[])
+    .filter(({resolve}) => !!resolve)
+    .reduce((arr, a) => arr.concat(a.resolve!.input), [] as string[])
     .forEach((ext) => needsDefaultPlugin.delete(ext));
   if (needsDefaultPlugin.size > 0) {
     plugins.unshift(esbuildPlugin({input: [...needsDefaultPlugin]}));
   }
 
+  const extensionMap = plugins.reduce((map, {resolve}) => {
```

Comment:
I guess I'm a bit confused about this part.

If `output` is optional, What happens if the user doesn't define `output`?
Should the input map to multiple extension types for outputs? 

Maybe I'm confusing 2 different parts that are using the terms `input` and `output`?